File: 78-subsets/78-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
            res = []
            subset = []
            def dfs(index):
                if index >= len(nums):
                    res.append(subset.copy())
                    return

                subset.append(nums[index])
                dfs(index + 1)
                subset.pop()
                dfs(index + 1)

            dfs(0)
            return res

        

File: 78-subsets/78-subsets.cpp
class Solution {
public:
    
    vector<vector<int>> ans ;

void dfs(vector<int>& nums ,vector<int> subset , int index){

	if(index == (int) nums.size()){
		ans.push_back(subset);
		return;
	}

	subset.push_back(nums[index]);
	dfs(nums , subset , index + 1);
	subset.pop_back();
	dfs(nums,subset , index+1);

}

    
    vector<vector<int>> subsets(vector<int>& nums) {
        	vector<int> subset;
	dfs(nums , subset , 0);

	return ans;
    }
};